gnovm/pkg/gnolang/machine.go

```diff
@@ -2131,25 +2131,25 @@ func (m *Machine) String() string {
 		totalLength = vsLength + ssLength + xsLength + bsLength + obsLength + fsLength + exceptionsLength
 	)
 
-	var builder strings.Builder
+	builder := new(strings.Builder)
```

Comment:
`strings.Builder` is a value type, so you don't need to create it using the new keyword. Using new will allocate memory on the heap, which may result in another unnecessary memory allocation.

Reply:
Not really, builder doesn't escape the heap. Go isn't like traditional languages where invoking "new" automatically allocates on the heap; escape analysis determines that and you can verifiably check that it doesn't escape and also run benchmarks to see that it doesn't escape using `-gcflags='-l -m'`. I can however bring it to a much better compromise of
```go
var sb strings.Builder
builder := &sb
```

which is a much better compromise than
```go
fmt.Fprintf(&sb, ...)
```
in every spot and materially even faster.

Reply:
Oh I didn't know that flag option has exists. Then this seems to have no problem at all. 👍